meteor/server/api/blueprints/context/syncIngestUpdateToPartInstance.ts

```diff
@@ -39,13 +55,54 @@ export class SyncIngestUpdateToPartInstanceContext extends RundownContext
 
 		this._partInstanceCache = new DbCacheWriteCollection(PartInstances)
 		this._partInstanceCache.fillWithDataFromArray([partInstance])
+
+		this._proposedPieceInstances = normalizeArrayToMap(proposedPieceInstances, '_id')
 	}
 
 	applyChangesToCache(cache: CacheForRundownPlaylist) {
 		this._pieceInstanceCache.updateOtherCacheWithData(cache.PieceInstances)
 		this._partInstanceCache.updateOtherCacheWithData(cache.PartInstances)
 	}
 
+	syncPieceInstance(
+		pieceInstanceId: string,
+		modifiedPiece?: Omit<IBlueprintPiece, 'lifespan'>
+	): IBlueprintPieceInstance {
+		const proposedPieceInstance = this._proposedPieceInstances.get(protectString(pieceInstanceId))
+		if (!proposedPieceInstance) {
+			throw new Error('PieceInstance could not be found')
```

Comment:
This error would benefit from adding the not-found pieceInstanceId. it should also be a Meteor.error